daos-stack/daos
# DAOS-10002 test: Add rank failure test

### Rank Failure Test
Run the following test steps with data protection (rf:1) with
RP_3GX and EC_4P2GX object class.

1. Create a pool and a container. Create a container with redundancy factor.
2. Run IOR with given object class and let it run through step 7.
3. While IOR is running, kill all daos_engine on a non-access-point node
4. Wait for IOR to complete.
5. Verify that IOR failed.
6. Wait for rebuild to finish.
7. Restart daos_servers.
8. Verify the system status by calling dmg system query.
9. Call dmg pool query -b to find the disabled ranks.
10. Call dmg pool reintegrate one rank at a time to enable all ranks.
11. Verify that the container Health is HEALTHY.
12. Run IOR and verify that it works.

### Rank Failure Isolation Test
Stop daos_engine where pool is not created.

1. Determine the two ranks to create the pool and a
node to kill the engines.
2. Create a pool across two ranks on the same node.
3. Create a container without redundancy factor.
4. Run IOR with oclass SX.
5. While IOR is running, kill daos_engine process
from two of the ranks where the pool isn’t created.
This will simulate the case where there’s a node
failure, but doesn’t affect the user because their
pool isn’t created on the failed node (assuming that
everything else such as network, client node, etc.
are still working).
6. Verify that IOR finishes successfully.
7. Verify that the container Health is HEALTHY.
8. To further verify that the pool isn’t affected,
create a new container on the pool and run IOR.
9. To make avocado happy, restart daos_servers on the
node where the engines were killed.

Skip-unit-tests: true
Test-tag: rank_failure

Signed-off-by: Makito Kano <makito.kano@intel.com>
